src/shell-window-tracker.c

```diff
@@ -435,6 +465,23 @@ get_app_for_window (ShellWindowTracker    *tracker,
   if (g_strcmp0 (meta_window_get_role (window), SIDE_COMPONENT_ROLE) == 0)
     return NULL;
 
+  /* We do want to associate the GNOME Builder window with
+   * an open app of a coding session */
+  if (g_strcmp0 (meta_window_get_flatpak_id (window), BUILDER_WINDOW) == 0)
+    {
+      if (tracker->coding_app)
+        {
+          result = g_hash_table_lookup (tracker->window_to_app, tracker->coding_app);
+          tracker->coding_app = NULL;
```

Comment:
Does not make a difference, but you could call the untrack function here for consistency